src/YelayStaking.sol

```diff
@@ -25,7 +23,7 @@ import "./interfaces/IRewardDistributor.sol";
  * At stake, gradual sYLAY (Yelay Voting Token) is minted and accumulated every week.
  * At unstake all sYLAY is burned. The maturing process of sYLAY restarts.
  */
-contract YelayStaking is ReentrancyGuardUpgradeable, YelayOwnable, IYelayStaking, EIP712 {
+contract YelayStaking is ReentrancyGuardUpgradeable, YelayOwnable, IYelayStaking {
```

Comment:
Since EIP712 does not define any storage variables or storage gaps, it can be safely removed from the inheritance chain.